### PR TITLE
DOC: Added docstring for show_versions()

### LINKED ISSUE #33044: Missing show_versions() in documentation
```python
pandas.show_versions()
```
is not in pandas API Reference.

### PR DESCRIPTION
**Pull request by MarianD**
- [x] closes #33044
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Comment by MarianD** [comment]
@TomAugspurger, connected with issue #33044

File: pandas/util/_print_versions.py [hunk 1]
```diff
@@ -88,6 +88,22 @@ def _get_dependency_info() -> Dict[str, JSONSerializable]:
...
+    ----------
+    as_json : str or bool, default False
+        * If False, outputs info in a human readable form to the console.
+        * If str, it will be considered as a path to a file.
+          Info will be written to that file in JSON format.
+        * If True, outputs info in JSON format to the console.
+
+    Returns
+    -------
+    None
```
Comment (TomAugspurger) [inline pandas/util/_print_versions.py]: I think this section can be omitted when there's no return value.

File: pandas/util/_print_versions.py [hunk 2]
```diff
@@ -88,6 +88,22 @@ def _get_dependency_info() -> Dict[str, JSONSerializable]:
 
 
 def show_versions(as_json: Union[str, bool] = False) -> None:
+    """
+    Provides information about hosting operation system, pandas version,
```
Comment (TomAugspurger) [inline pandas/util/_print_versions.py]: Needs to be a single line.

**Review by TomAugspurger — COMMENTED** [review]
Thanks! Can you run `scripts/validate_docstrings.py` on this? I don't think we need any of the other sections, just the Parameters.

You can also add it to the API docs. Probably in `doc/source/reference/general_utility_functions.rst`.

**Comment by TomAugspurger** [comment]
You can ignore the CI failure.

**Comment by MomIsBestFriend** [comment]
@MarianD Can you please merge master?

**Comment by MarianD** [comment]
> 
> 
@MomIsBestFriend 
> @MarianD Can you please merge master?

I merged my “docstring” branch into “master” in my fork and created pull request for that branch (master) — is it what you expected?

**Comment by MomIsBestFriend** [comment]
> > 
> 
> @MomIsBestFriend
> 
> > @MarianD Can you please merge master?
> 
> I merged my “docstring” branch into “master” in my fork and created pull request for that branch (master) — is it what you expected?

@MarianD sorry this is not what I expected, but it's perfectly fine.

I should have been more specific, what I meant was "please update your pull request"

https://dev.pandas.io/docs/development/contributing.html#updating-your-pull-request

This is necessary due to a failure we had in our CI (continues interrogation), and updating your PR (pull request) would solve this.

File: doc/source/reference/general_utility_functions.rst [hunk 3]
```diff
@@ -30,6 +30,13 @@ Testing functions
    testing.assert_index_equal
    testing.assert_extension_array_equal
 
+Bug report function
```
Comment (TomAugspurger) [inline doc/source/reference/general_utility_functions.rst]: I'd put this at the bottom of this file.

**Review by TomAugspurger — COMMENTED** [review]
Looks like there's a linting issue: https://github.com/pandas-dev/pandas/pull/33073/checks?check_run_id=543334674#step:10:9

File: doc/source/reference/general_utility_functions.rst (hunk 3 above)
Comment (MarianD) [reply doc/source/reference/general_utility_functions.rst]: @TomAugspurger, I did it.

**Review by datapythonista — APPROVED** [review]
lgtm, thanks @MarianD 

ping @TomAugspurger

**Review by TomAugspurger — APPROVED** [review]

**Comment by TomAugspurger** [comment]
Thanks1

**Pull request merged**